job_templates/scripts/add_recovery_label_on_regex.groovy

```diff
@@ -0,0 +1,25 @@
+  if(!regex) {
+    return """println('WARNING! Incorrect recovery configuration : No regex provided as search key for log')"""
+  }
+  return """\
+    import hudson.model.*
+    def node = build.getBuiltOn()
+    def oldLabels = node.getLabelString()
+
+    println('# BEGIN SECTION: AGENT RECOVERY PROCESS')
+    if (!(build.getLog(1000) =~ ${regex})) {
+         println('${error} not detected in the log - Not performing any recovery automatic recovery step')
+         return 1;
+    } else {
+     try {
+         println(' PROBLEM: ${error} was detected in the log. Try to automatically resolve it:')
+         println("Removing labels and adding 'recovery-process' label to node")
+         node.setLabelString('recovery-process')
+         } catch (Exception ex) {
+         println('ERROR - CANNOT PERFORM RECOVERY ACTIONS FOR NVIDIA ERROR')
+         println('Restoring to previous state') node.setLabelString(oldLabels)
+         throw ex
+     }
+    }
+    println('# END SECTION: NVIDIA MISMATCH RECOVERY')
```

Comment:
This looks like it was copy-pasted from the build.osrfoundation.org recovery without updating the section output.

I'm also not sure that this script needs to be wrapped in a script since the Groovy DSL isn't used.